#! /bin/sh -e
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Improve security check (temporary patch, already merged upstream)

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch)   ( patch $patch_opts -p0 ) < $0 ;;
       -unpatch) ( patch $patch_opts -p0 -R ) < $0 ;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@
--- login.pl	2006-11-24 23:11:23.000000000 +0100
+++ login.pl.new	2006-11-27 08:27:20.000000000 +0100
@@ -71,7 +71,7 @@
 
 @scripts = qw(login.pl admin.pl custom_login.pl custom_admin.pl);
 
-if (grep !/$form{script}/, @scripts) {
+if (grep !/^\Q$form{script}\E/, @scripts) {
   print "Content-Type: text/html\n\n" if $ENV{HTTP_USER_AGENT};
   print "\nAccess denied!\n";
   exit;
